.buildkite/baristacli.py

```diff
@@ -0,0 +1,569 @@
+#     "name": "AWS Linux 2", # label
+#     "platform": "linux"    # specify platform (linux/windows)
+#   },
+# }
+PLATFORMS = {
+    "default": {
+        "name": ":linux: AWS Linux 2",
+        "platform": "linux"
+    },
+    "windows10": {
+        "name": ":windows: Windows 10",
+        "platform": "windows"
+    }
+}
+
+class BuildKiteConfigError(ValueError):
+    """ BuildKiteConfigError Exception """
+
+class BazelFailingTests(Exception):
+    """ BazelFailingTests Exception"""
+
+
+def is_windows():
+    """
+    Return True if we are on windows.
+    """
+    return os.name == "nt"
+
+
+# set script extension depending on platform
+SCRIPT_EXTENSION = ".sh" if not is_windows() else ".bat"
+
+
+def eprint(*args, **kwargs):
+    """
+    Print to stderr and flush (just in case).
+    """
+    print(*args, flush=True, file=sys.stderr, **kwargs)
+
+
+def execute_shell_commands(commands):
+    """
+    Prepare given list of commands and execute it
+    :param commands: array if commands
+    """
+    if not commands:
+        return
+    if is_windows():
+        print_collapsed_group(":cmd: Setup (Windows Commands)")
+        for cmd in commands:
+            execute_command(cmd.split(" "), shell=True)
+    else:
+        print_collapsed_group(":bash: Setup (Shell Commands)")
+        shell_command = "\n".join(["set -e"] + commands)
+        execute_command([shell_command], shell=True)
+
+
+def execute_command(args, shell=False, fail_if_nonzero=True, cwd=None,
+                    print_output=True):
+    """
+    Execute command
+    """
+    if print_output:
+        eprint(" ".join(args))
+    return subprocess.run(
+        args, shell=shell, check=fail_if_nonzero, env=os.environ, cwd=cwd
+    ).returncode
+
+
+def arg_hander_command(args):
+    """
+    Prepare the given platform and task to be able to call it
+    """
+    tasks = load_config(args.config)
+
+    if not args.platform in tasks["tasks"].keys():
+        raise BuildKiteConfigError(
+            "{} is not available. Please specify one of {}".format(args.platform,
+                                                                   PLATFORMS.keys()))
+
+    stage = tasks["tasks"].get(args.platform)
+
+    if not args.target in stage:
+        raise BuildKiteConfigError(
+            "{} is not available. Please specify one of {}".format(args.target,
+                                                                   stage.keys()))
+
+    # execute pre commands
+    if "pre_cmd" in stage[args.target]:
+        execute_shell_commands(stage[args.target]['pre_cmd'])
+
+    # bazel build info file
+    build_event_file = "build_event_file.json"
+
+    if args.target == "test" and 'parallel' in stage[args.target] and int(
+            args.shard) != 0:
+        bazel_flags = stage[args.target]["bazel_flags"] if "bazel_flags" in stage[
+            args.target] else []
+        # add build event json
+        bazel_flags += ["--build_event_json_file={}".format(build_event_file)]
```

Comment:
I would add for every bazel command the `--profile=./profile.gz --record_full_profiler_data` build profile. and store it as gz. This is the holy grail if you want to investigate performance issues in the build. would be nice if we can store it as artifact to download it. you can upload the file to `chrome://tracing` and get insights in your build.

Reply:
![image](https://user-images.githubusercontent.com/43139367/95571646-13611b00-0a29-11eb-898e-6dc23fb67422.png)